platform/mds/mds/src/test/java/org/motechproject/mds/it/service/EntityServiceContextIT.java

```diff
@@ -215,6 +215,32 @@ public void shouldSaveEntityWithGivenLookups() throws IOException {
     }
 
     @Test
+    public void shouldAddNewFieldForLookupAndSaveEntity() throws IOException {
+        EntityDto entityDto = new EntityDto();
+        entityDto.setName("myEntity");
+        entityDto = entityService.createEntity(entityDto);
+
+        EntityDraft entityDraft = entityService.getEntityDraft(entityDto.getId());
+        LookupDto lookup = new LookupDto("lookup", false, false, null, true);
+        entityService.addLookups(entityDraft.getId(), Collections.singletonList(lookup));
+        entityService.saveDraftEntityChanges(entityDto.getId(), DraftBuilder.forNewField("disp", "testFieldName", Long.class.getName()));
+        FieldDto field = selectFirst(entityService.getFields(entityDto.getId()), having(on(FieldDto.class).getBasic().getName(), equalTo("testFieldName")));
```

Comment:
group this once again, after creating an entity, create field and then start with lookup